packages/react-instantsearch-core/src/widgets/Index.tsx

```diff
@@ -99,18 +92,10 @@ class Index extends Component<InnerProps, State> {
     );
   }
 
-  componentWillReceiveProps(nextProps: InnerProps) {
-    // @TODO: DidUpdate
-    if (this.props.indexName !== nextProps.indexName) {
+  componentDidUpdate(prevProps: InnerProps) {
+    if (this.props.indexName !== prevProps.indexName) {
```

Comment:
We might have the same issue raised inside the PR #2349 about the duplicate request. It's not easy to test right now because the multi-index API does not work anymore. We should fix it first, then test that we do have the correct behaviour.

Reply:
Fixed in https://github.com/algolia/react-instantsearch/pull/2552 , should be ok now.